File: sivitas.py
from mahasiswa import Mahasiswa
from dosen import Dosen

# Kelas SivitasAkademik sebagai induk
class SivitasAkademik:
    
    # access modifier private
    __asal_universitas = ""
    __email_edu = ""

    # constructor
    def __init__(self, asal_universitas = "", email_edu = ""):
        self.__asal_universitas = asal_universitas
        self.__email_edu = email_edu

    def get_asal_universitas(self):
        return self.__asal_universitas
    
    def set_asal_universitas(self, asal_universitas):
        self.__asal_universitas = asal_universitas

    def get_email_edu(self):
        return self.__email_edu
    
    def set_email_edu(self, email_edu):
        self.__email_edu = email_edu

    #fungsi untuk menambahkan isi array, note: tapi tidak terpakai
    def __str__(self):
        lines = []
        lines.append(f'{self.asal_universitas}, {self.email_edu}')
        return '\n'.join(lines)

File: cpp/sivitas.cpp
//import library
#include <bits/stdc++.h>
#include "mahasiswa.cpp"
#include "dosen.cpp"

//using standard namespace
using namespace std;
//SivitasAkademik sebagai class induk
class SivitasAkademik
{
    //access modifier private
    private:
        string asal_universitas;
        string email_edu;
    //access modifier publlik
    public:
        //constructor
        SivitasAkademik()
        {
            this->asal_universitas = " ";
            this->email_edu = " ";
        }

        //penggunaan getter dan setter
        string get_asal_universitas()
        {
            return this->asal_universitas;
        }

        void set_asal_universitas(string asal_universitas)
        {
            this->asal_universitas = asal_universitas;
        }

        string get_email_edu()
        {
            return this->email_edu;
        }

        void set_email_edu(string email_edu)
        {
            this->email_edu = email_edu;
        }

        ~SivitasAkademik()
        {

        }

};




